compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/AddWebSocketCodeTemplate.java

```diff
@@ -0,0 +1,135 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.websocket.plugin;
+
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.ServiceDeclarationNode;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.projects.plugins.codeaction.CodeAction;
+import io.ballerina.projects.plugins.codeaction.CodeActionArgument;
+import io.ballerina.projects.plugins.codeaction.CodeActionContext;
+import io.ballerina.projects.plugins.codeaction.CodeActionExecutionContext;
+import io.ballerina.projects.plugins.codeaction.CodeActionInfo;
+import io.ballerina.projects.plugins.codeaction.DocumentEdit;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticProperty;
+import io.ballerina.tools.text.LineRange;
+import io.ballerina.tools.text.TextDocument;
+import io.ballerina.tools.text.TextDocumentChange;
+import io.ballerina.tools.text.TextEdit;
+import io.ballerina.tools.text.TextRange;
+import org.wso2.ballerinalang.compiler.diagnostic.properties.NonCatProperty;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Code action to add resource code snippet.
+ */
+public class AddWebSocketCodeTemplate implements CodeAction {
+
+    public static final String NODE_LOCATION = "node.location";
+    public static final String RESOURCE_TEXT = "\n\tresource function get .() returns " +
+            "websocket:Service|websocket:Error " +
+            "{\n\t\treturn new WsService(); \n\t}\n";
+    public static final String SERVICE_TEXT = "\n\nservice class WsService {\n" +
+            "\t*websocket:Service;\n\n" +
+            "\tremote isolated function onTextMessage(websocket:Caller caller, string text) " +
+            "returns websocket:Error? {\n" +
+            "\t}\n" +
+            "}";
+
+    @Override
+    public List<String> supportedDiagnosticCodes() {
+        return List.of(PluginConstants.CompilationErrors.TEMPLATE_CODE_GENERATION_HINT.getErrorCode());
+    }
+
+    @Override
+    public Optional<CodeActionInfo> codeActionInfo(CodeActionContext codeActionContext) {
+        Diagnostic diagnostic = codeActionContext.diagnostic();
+        List<DiagnosticProperty<?>> properties = diagnostic.properties();
+        if (properties.isEmpty()) {
+            return Optional.empty();
+        }
+        DiagnosticProperty<?> diagnosticProperty = properties.get(0);
+        if (!(diagnosticProperty instanceof NonCatProperty) ||
```

Comment:
Instead, let's use the `diagnostic.location()` to find the node from `SyntaxTree` and use it. No need to use the diagnostic properties.